controllers/csi/gc/logs.go

```diff
@@ -0,0 +1,124 @@
+package csigc
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	dtcsi "github.com/Dynatrace/dynatrace-operator/controllers/csi"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+const (
+	maxLogFolderSize    = 300000
+	maxAmountOfLogFiles = 1000
+	maxLogAge           = 14 * 24 * time.Hour
+)
+
+func (gc *CSIGarbageCollector) runLogGarbageCollection(versionReferences []os.FileInfo, tenantUUID string) error {
+	fs := &afero.Afero{Fs: gc.fs}
+	gcPodUIDs, err := gc.getPodUIDs(fs, versionReferences, tenantUUID)
+	if err != nil {
+		gc.logger.Info("skipped, failed to get podUIDs")
+		return errors.WithStack(err)
+	}
+
+	logPath := filepath.Join(gc.opts.RootDir, dtcsi.DataPath, tenantUUID, dtcsi.LogDir)
+	logPodUIDs, err := fs.ReadDir(logPath)
+	if err != nil {
+		gc.logger.Info("skipped, failed to get references")
+		return errors.WithStack(err)
+	}
+
+	deadPods := difference(gcPodUIDs, logPodUIDs)
+
+	shouldDelete := len(deadPods) > 0 && (sizeTooBig(fs, logPath) || len(deadPods) > maxAmountOfLogFiles)
+	if shouldDelete {
+		err = gc.removeDeadLogFolders(deadPods, fs, logPath)
+		if err != nil {
+			gc.logger.Error(err, "failed to remove dead log directories")
+			return errors.WithStack(err)
+		}
+	}
+
+	return nil
+}
+
+func (gc *CSIGarbageCollector) removeDeadLogFolders(deadPods []os.FileInfo, fs *afero.Afero, logPath string) error {
```

Comment:
Suggest to rename it to something like `tryRemoveLogFolders`. It checks some condition before a folder gets actually deleted.

Reply:
Renamed